doc/cascadia/Cascading-Default-Settings.md

```diff
@@ -0,0 +1,571 @@
+never roam to machines where they don't exist and aren't valid, but the
+generators _must_ be enabled to use the dynamic profiles.
+
+## Future considerations
+* It's possible that a very similar layering loading mechanism could be used to
+  layer per-machine settings with roaming settings. Currently, there's only one
+  settings file, and it roams to all your devices. This could be problematic,
+  for example, if one of your machines has a font installed, but another
+  doesn't. A proposed solution to that problem was to have both roaming settings
+  and per-machine settings. The code to layer settings from the defaults and the
+  user settings could be re-used to handle layer the roaming and per-machine
+  settings.
+* What if an extension wants to generate their own dynamic profiles? We've
+  already outlined a contract that profile generators would have to follow to
+  behave correctly. It's possible that we could abstract our implementation into
+  a WinRT interface that extensions could implement, and be triggered just like
+  other dynamic profile generators.
+  - Additionally, we could expand the dynamic profile generator contract to
+    force it to use a "namespace GUID". We'd then have the generator ask us to
+    generate a profile GUID for some static string, given it's namespace GUID.
+    The generator then wouldn't need to always generate unique GUIDs for its
+    profiles, but it would need to generate unique strings. We already allow
+    the user to disable profile generators solely by namespace guid, so they
+    could disable the generator from a certain extension easily.
+* **Multiple settings files** - This could enable us to place color schemes into
+  a seperate file (like `colorschemes.json`) and put keybindings into their own
+  file as well, and reduce the number of settings in the user's `profiles.json`.
+  It's unclear if this is something that we need quite yet, but the same
+  layering functionality that enables this scenario could also enable more than
+  two sources for settings.
+* **Global Default Profile Settings** - Say a user wants to override what the
+  defaults for a profile are, so that they can set settings for _all_ their
+  profiles at once? We could maybe introduce a profile in the user settings file
+  with a special guid set to `"default`, that we look for first, and treat
+  specially. We wouldn't include it in the list of profiles. When we're creating
+  profiles, we'll start with that profile as our prototype, instead of using the
+  default-constructed `Profile`. When we're serializing profiles, we'd again use
+  that as the point of comparison to check if a setting's value has changed.
+  There may be more unknowns with this proposal, so I leave it for a future
+  feature spec.
+  - How do these play with dynamically generated profiles? If the user sets a
+    `commandline` for their global default profile, then the dynamic profile

```

Comment:
they just shouldn't be able to do that, i dunno

Reply:
Yea I'm just pulling this out now, since it has kinda been addressed earlier, and we're not relying upon `commandline` to see if a profile is valid anymore.